hydra_base/lib/scenario.py

```diff
@@ -1336,6 +1337,40 @@ def get_attribute_data(attr_ids, node_ids, **kwargs):
 
     return node_attrs, resource_scenarios
 
+def _get_all_network_resource_attributes(network_id):
+    """
+        Get all the attributes for the nodes, links and groups of a network.
+        Return these attributes as a dictionary, keyed on type (NODE, LINK, GROUP)
+        then by ID of the node or link.
+
+        args:
+            network_id (int) The ID of the network from which to retrieve the attributes
+        returns:
+            A list of sqlalchemy result proxy objects
+    """
+    base_qry = db.DBSession.query(ResourceAttr).filter(Attr.id==ResourceAttr.attr_id)
+
+    all_node_attribute_qry = base_qry.join(Node).filter(Node.network_id == network_id)
+
+    all_link_attribute_qry = base_qry.join(Link).filter(Link.network_id == network_id)
+
+    all_group_attribute_qry = base_qry.join(ResourceGroup)\
+            .filter(ResourceGroup.network_id == network_id)
+
+    network_attribute_qry = base_qry.filter(ResourceAttr.network_id == network_id)
+
+
+    x = time.time()
+    logging.info("Getting all attributes using execute")
+    attribute_qry = all_node_attribute_qry.union(all_link_attribute_qry,
+                                                 all_group_attribute_qry,
+                                                 network_attribute_qry)
+    all_resource_attributes = attribute_qry.all()
+
+    log.info("%s attrs retrieved in %s", len(all_resource_attributes), time.time()-x)
```

Comment:
Extra pedantic: doesn't give unit (seconds) and has potentially large number of decimal places, e.g "2.474357862952398".

Can use `round(time.time()-x, 2)` to avoid changing string and formatting.